### PR TITLE
feat: add linea network support [APE-1170]

### PR DESCRIPTION
**Pull request by dawnseeker8**
### What I did

Add the Linea network (Mainnet and goerli testnet) support to the plugin , 

### How I did it
* add linea network information to plugin
* add new test in test.

### How to verify it
```
pip install ape-linea
ape networks list
ape console --network linea:goerli:infura
ape console --network linea:mainnet:infura
```

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/test_provider.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
         ("optimism", "goerli"),
         ("polygon", "mainnet"),
         ("polygon", "mumbai"),
+        ["linea", "mainnet"],
```
Comment (antazoey) [inline tests/test_provider.py:20]:
```suggestion
        ("linea", "mainnet"),
```

File: tests/test_provider.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
...
+        ["linea", "goerli"],
```
Comment (antazoey) [inline tests/test_provider.py:21]:
```suggestion
        ("linea", "goerli"),
```

**Review by antazoey — COMMENTED** [review]
One small inconsistency

File: tests/test_provider.py (hunk 1 above)
Comment (dawnseeker8) [reply tests/test_provider.py]: sorry, my mistake, have changed to use ();

**Review by antazoey — APPROVED** [review]

**Pull request merged**